lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2536,25 +2536,6 @@ reapplyFunctionConversion(
   llvm_unreachable("Unhandled function conversion instruction");
 }
 
-static SubstitutionMap getSubstitutionMap(
```

Comment:
More context: I added this `getSubstitutionMap` utility as a hack in https://github.com/apple/swift/pull/24845, I don't think the old behavior was necessary. The new simpler behavior seems sufficient for now.